src/java/org/apache/cassandra/utils/SyncUtil.java

```diff
@@ -99,6 +101,12 @@
     public static MappedByteBuffer force(MappedByteBuffer buf)
     {
         Preconditions.checkNotNull(buf);
+        Object attachment = MemoryUtil.getAttachment(buf);
+        if (attachment instanceof Runnable)
```

Comment:
the `FileSystem` runs in app `ClassLoader` but this logic runs in `InstanceClassLoader`, so I can't use a non standard type here...

This is a field access, but if there are concerns about this I can always make this a test flag, so we only enter this if a static field is set to true

Reply:
What touches this? The commitlog? I suppose erring on the side of caution would be okay...

Reply:
listenable fs... we need a way to fake sync, and C* detects that we don't have a mmap buffer and rejects